content/sensu-go/5.10/reference/datastore.md

```diff
@@ -0,0 +1,164 @@
+---
+title: "Datastore"
+description: "Sensu stores the most recent event for each entity and check pair using an embedded etcd (default) or an external etcd instance. Sensu also supports using an external PostgreSQL instance for event storage in place of etcd (licensed-tier only). Read the reference to configure enterprise-scale event storage using PostgreSQL."
+weight: 10
+version: "5.10"
+product: "Sensu Go"
+menu:
+  sensu-go-5.10:
+    parent: reference
+---
+
+- [Event storage](#event-storage)
+- [Scaling event storage](#scaling-event-storage) (licensed-tier only)
+  - [Requirements](#requirements)
+  - [Configuration](#configuration)
+  - [Specification](#specification)
+
+## Event storage
+
+Sensu stores the most recent event for each entity and check pair using an embedded etcd (default) or an [external etcd][8] instance.
+You can access event data using the [dashboard][9] events page, [`sensuctl event` commands][10], and the [events API][11].
+To store event data over time, we recommend integrating Sensu with a time series database, like [InfluxDB][12].
```

Comment:
```suggestion
For longer retention of event data, we recommend integrating Sensu with a time series database, like [InfluxDB][12], or a searchable index, like ElasticSearch or Splunk.
```

Reply:
Thanks, @cwjohnston! Can you elaborate on this suggestion?

Reply:
In most cases where I've seen folks integrating Sensu with tsdb for extended event storage, the data stored in a tsdb tends to be a very lossy representation of the event. Typically this is the timestamp and a selection of attributes such as the check name, entity name and check result status. 

This is usually enough context for overlaying monitoring events on graphs which illustrate related metrics/telemetry, and in many cases this is probably a reasonable tradeoff, as I imagine adding a large number of labels/tags to each measurement isn't a recommended practice.

On the other hand, indexers like elasticsearch and splunk are typically considered a reasonable technology for recording complete event records. Right now this should be doable using our licensed feature for recording events to a flat file, using something like ElasticSearch filebeat or the Splunk log collector to store that data. In recent years these indexers have made strides to become better data sources for metrics/telemetry, making it possible that a team could choose to adopt one of these instead or or in addition to a timeseries database.

Reply:
This is great context, thanks!